### PR TITLE
Feature/715 forward xauth token to cprs

### PR DESCRIPTION
**Pull request by kzangeli**
o Func test for forwarding of X-Auth tokens in HTTP header (Issue #715)
o Made the broker not forward Service-Path if no Service-Path has been issued in the initial request (Issue #714)

### CODE REVIEW

File: scripts/accumulator-server.py [hunk 1]
```diff
@@ -115,6 +114,18 @@ def all_done():
...
+    global ac
+    ac += "Got a /v1/updateContext";
+    return Response(status=200)
+
+@app.route("/v1/queryContext", methods=['POST'])
+def record():
+    global ac
+    ac += "Got a /v1/queryContext";
+    return Response(status=200)
+
```
Comment (kzangeli) [inline scripts/accumulator-server.py]: Now tell me why this 'stupid' modification makes the accumulator behave the way I want ...
I expected to see 'Got a /v1/updateContext' and 'Got a /v1/queryContext' in the dump, and after that keep working to get the message in 'ac' instead.
However, in the first test I got already the exact output I wanted from /dump. Magic ... ?

Also, I've seen that in some functions use declare 'ac' as global and in others you don't - confusing.

File: src/lib/rest/rest.cpp [hunk 2]
```diff
@@ -789,7 +789,6 @@ static int connectionTreat
     if (!ciP->httpHeaders.servicePathReceived)
     {
       ciP->httpHeaders.servicePath = defaultServicePath(url, method);
-      ciP->httpHeaders.servicePathReceived = true;
```
Comment (kzangeli) [inline src/lib/rest/rest.cpp]: This here was a mistake from not so long ago.
I saw it in the PR but I let it pass, didn't think it mattered.
However, for this fix I need to know whether the servicePath is actually path of the request or not, so I had to remove it.

Just informative.
NTC

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -7,3 +7,4 @@
 - Fix: Service-Path supported for forwarding of registrations between Config Manager and Context Broker (Issue #719)
 - Fix: Service-Path supported for forwarding to Context Providers (Issue #719)
 - Fix: Hash sign (#) in service-path, or multiple service-paths not valid for updateContext requests (Issue #746)
+- Fix: Not forwarding the default Service-Path to context providers (Issue #714)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: I understand that in the case of Service header everything is ok from time ago, right?

Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Maybe we need a .test (in the 714_* directory) to ensure that service path is not being forwarded in this case?

File: scripts/accumulator-server.py (hunk 1 above)
Comment (fgalan) [inline scripts/accumulator-server.py]: What is even more weird is  having three times the same function defined, i.e. `record()`... maybe the `@app.route` is taken into account in the function definition and makes signatures different.

It seems that the function that is actually called is the last instance of `record()`. That would explain what it works and you don't have the "Got blabla" messages in the /dump.

Try to remove the first two function instances and set th following in the last one:

```
@app.route("/v1/updateContext", methods=['POST'])
@app.route("/v1/queryContext", methods=['POST'])
@app.route(server_url, methods=['GET', 'POST', 'PUT', 'DELETE'])
def record():
    global ac, t0, times
    ...
```

File: test/functionalTest/cases/715_forward_xauth_token_to_cprs/propagate_xauth_token_for_update_and_query_standard_ops.test [hunk 4]
```diff
@@ -0,0 +1,216 @@
+# Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/functionalTest/cases/715_forward_xauth_token_to_cprs/propagate_xauth_token_for_update_and_query_standard_ops.test]: Maybe we should add a a .test to ensure that X-Auth-Token is not being forwarded when X-Auth-Token is not included in the request? Or existing tests are covering that functionality?

File: CHANGES_NEXT_RELEASE (hunk 3 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Nope, it works, but it is *not* covered in tests, also not that a query/update without X-Auth token doesn't send an X-Auth token in the forward.

File: test/functionalTest/cases/715_forward_xauth_token_to_cprs/propagate_xauth_token_for_update_and_query_standard_ops.test (hunk 4 above)
Comment (kzangeli) [inline test/functionalTest/cases/715_forward_xauth_token_to_cprs/propagate_xauth_token_for_update_and_query_standard_ops.test]: Yes, as I just responded to the previous comment :-)
New func-tests are in the oven.

Comment (kzangeli) [inline test/functionalTest/cases/715_forward_xauth_token_to_cprs/propagate_xauth_token_for_update_and_query_standard_ops.test]: Fixed in 27668ef

File: CHANGES_NEXT_RELEASE (hunk 3 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in 5e51d1e

File: scripts/accumulator-server.py (hunk 1 above)
Comment (kzangeli) [inline scripts/accumulator-server.py]: Fixed in 5e51d1e

**Comment by fgalan** [comment]
LGTM

**Pull request merged**